pandas/core/generic.py

```diff
@@ -3128,10 +3128,12 @@ def to_csv(
 
         Parameters
         ----------
-        path_or_buf : str or file handle, default None
-            File path or object, if None is provided the result is returned as
-            a string.  If a file object is passed it should be opened with
-            `newline=''`, disabling universal newlines.
+        path_or_buf : str, path object, file-like object, default None
+            If a string is passed, it can be an existing local file path, as
+            well as a gcs:// or s3:// path. If a file object is passed it
```

Comment:
So I might be a little too strict, but I'm generally against documenting requirements of third party dependencies - would these not generally fall under the umbrella of being "file-like" to begin with?